api/python/cellxgene_census/src/cellxgene_census/experimental/ml/pytorch.py

```diff
@@ -131,34 +131,33 @@ def __next__(self) -> ObsDatum:
         while len(obs) < self.batch_size:
             try:
                 obs_partial, X_partial = self._read_partial_torch_batch(self.batch_size)
-                if X is None:
-                    obs = obs_partial
-                    X = X_partial
-                else:
-                    obs = pd.concat([obs, obs_partial], axis=0)
-                    X = sparse.vstack([X, X_partial])
-
+                obs = pd.concat([obs, obs_partial], axis=0)
+                X = sparse.vstack([X, X_partial])
             except StopIteration:
                 break
 
         if len(obs) == 0:
             raise StopIteration
 
+        assert obs["soma_joinid"].is_unique
```

Comment:
Mostly a nitpick, but I think this operation incurs into additional cost since determining uniqueness on a non-sorted array can't be done in linear time. Unless this is part of the operations flow, it might be wise to remove this.